x-pack/plugins/apm/server/routes/correlations/queries/fetch_duration_percentiles.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  SPAN_DURATION,
+  TRANSACTION_DURATION,
+} from '../../../../common/elasticsearch_fieldnames';
+import { SIGNIFICANT_VALUE_DIGITS } from '../../../../common/correlations/constants';
+import { Setup } from '../../../lib/helpers/setup_request';
+import { ProcessorEvent } from '../../../../common/processor_event';
+import { getCommonCorrelationsQuery } from './get_common_correlations_query';
+import { CommonCorrelationsQueryParams } from '../../../../common/correlations/types';
+
+export const fetchDurationPercentiles = async ({
+  eventType,
+  setup,
+  start,
+  end,
+  environment,
+  kuery,
+  query,
+  percents,
+}: CommonCorrelationsQueryParams & {
+  eventType: ProcessorEvent;
+  setup: Setup;
+  percents?: number[];
+}): Promise<{
+  totalDocs: number;
+  percentiles: Record<string, number>;
+}> => {
+  const response = await setup.apmEventClient.search(
+    'get_duration_percentiles',
+    {
+      apm: { events: [eventType] },
+      body: {
+        track_total_hits: true,
+        query: getCommonCorrelationsQuery({
+          start,
+          end,
+          environment,
+          kuery,
+          query,
+        }),
+        size: 0,
+        aggs: {
+          duration_percentiles: {
+            percentiles: {
+              hdr: {
+                number_of_significant_value_digits: SIGNIFICANT_VALUE_DIGITS,
+              },
+              field:
+                eventType === ProcessorEvent.span
+                  ? SPAN_DURATION
+                  : TRANSACTION_DURATION,
+              ...(Array.isArray(percents) ? { percents } : {}),
+            },
+          },
+        },
+      },
+    }
+  );
+
+  // return early with no results if the search didn't return any documents
```

Comment:
IMHO this comment is not necessary.

Reply:
Nah, but I don't feel strong enough about it to remove it either.